### ISSUE TITLE
Accessibility: WorkflowStepper Progress Indicator

### ISSUE THREAD

**Issue by jaydonkrooss** [tl-its-umich-edu/canvas-course-manager-next#701]
Problematic component: `WorkflowStepper.tsx`

Issue summary: 
- Progress indicator not announcing: missing ARIA on stepper

Based on these reported problems found in [CCM Accessibility Review](https://docs.google.com/spreadsheets/d/1mYrbqwFLLXQoGRTY_XEGkfS3cu3BO_pUN_zQ_DFdhUY/edit?gid=1709548063#gid=1709548063):

- [Severity: Critical] Add U-M Users One Section
    - Screen reader did not announce the progress indicator or indicate that is a part of "Select - Upload - Review - Confirmation"
- [Severity: Critical] Add Multiple Non-UM Users
    - Screen reader did not announce the progress indicator or indicate that is a part of "Select - Upload - Review - Confirmation"
- [Severity: Critical] Format 3rd Party Gradebook
    - Screen reader did not announce the progress indicator or indicate that is a part of "Select - Upload - Review - Confirmation"

<img width="1048" height="133" alt="Image" src="https://github.com/user-attachments/assets/afb21bb6-b66b-41d0-99d2-a595837c3be2" />
